src/init.cc

```diff
@@ -722,7 +722,11 @@ static ncclResult_t fillInfo(struct ncclComm* comm, struct ncclPeerInfo* info, u
 #endif
     CUDACHECK(hipFree(ptr));
     info->hasFineGrain = true;
-    NCCLCHECK(ncclGpuGdrSupport(comm, &info->gdrSupport));
+    // GPU supports GDR if DMABUF is supported
+    if (dmaBufSupported(comm) == ncclSuccess)
```

Comment:
Not your changes, but it seems strange that `dmaBufSupported` returns `ncclSuccess` when it means it is supported. That's not a success case, it's a affirmative (`true`) case.

Reply:
this function is from NCCL and we didn't change how it behaves